src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1636,18 +1636,17 @@ def _get_log(url, user_name, password, log_file=None):
 def upload_ssl_cert(cmd, resource_group_name, name, certificate_password, certificate_file):
     client = web_client_factory(cmd.cli_ctx)
     webapp = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get')
-    cert_resource_group_name = parse_resource_id(webapp.server_farm_id)['resource_group']
     cert_file = open(certificate_file, 'rb')
     cert_contents = cert_file.read()
     hosting_environment_profile_param = (webapp.hosting_environment_profile.name
                                          if webapp.hosting_environment_profile else '')
 
     thumb_print = _get_cert(certificate_password, certificate_file)
     cert_name = _generate_cert_name(thumb_print, hosting_environment_profile_param,
-                                    webapp.location, cert_resource_group_name)
```

Comment:
Will this regress the scenario that the ASP comes from an external RG? The old logic indicates the cert is bound to the ASP. If yes, then the code change should be checking the ASP's sku first and only invoke the new logic when the plan shows as ASE?

Reply:
Let me test this (ASP in different RG) out explicitly & comment on this thread.

Reply:
@yugangw-msft  i did test this with the latest changes & it does work in the ASP & App in different Resource groups as well